worlds/kh2/__init__.py

```diff
@@ -0,0 +1,314 @@
+
+            for location in self.multiworld.exclude_locations[self.player].value:
+                if location in self.RandomSuperBoss:
+                    self.RandomSuperBoss.remove(location)
+            #  Testing if the player has the right amount of Bounties for Completion.
+            if len(self.RandomSuperBoss) < self.BountiesAmount:
+                print(f"{self.multiworld.get_file_safe_player_name(self.player)} has too many bounties than bosses")
+                self.BountiesAmount = len(self.RandomSuperBoss)
+                self.multiworld.BountyAmount[self.player].value = len(self.RandomSuperBoss)
+
+            if len(self.RandomSuperBoss) < self.BountiesRequired:
+                print(f"{self.multiworld.get_file_safe_player_name(self.player)} has too many required bounties")
+                self.BountiesRequired = len(self.RandomSuperBoss)
+                self.multiworld.BountyRequired[self.player].value = len(self.RandomSuperBoss)
+
+            if self.BountiesAmount < self.BountiesRequired:
+                self.BountiesAmount = max(self.BountiesAmount, self.BountiesRequired)
+                print(f"Bounties Amount {self.multiworld.BountyAmount[self.player].value} is less than required \
+                        {self.multiworld.BountyRequired[self.player].value} for player {self.multiworld.get_file_safe_player_name(self.player)}")
+            self.multiworld.start_hints[self.player].value.add(ItemName.Bounty)
+            self.item_quantity_dict.update({ItemName.ProofofNonexistence: 0})
+
+        while len(self.sora_keyblade_ability_pool) < len(self.keyblade_slot_copy):
+            self.sora_keyblade_ability_pool.append(
+                self.multiworld.per_slot_randoms[self.player].choice(list(SupportAbility_Table.keys())))
+
+        for item in DonaldAbility_Table.keys():
+            data = self.item_quantity_dict[item]
+            for x in range(data):
+                self.donald_ability_pool.append(item)
+            self.item_quantity_dict.update({item: 0})
+            # 32 is the amount of donald abilities
+        while len(self.donald_ability_pool) < 32:
+            self.donald_ability_pool.append(self.multiworld.per_slot_randoms[self.player].choice(self.donald_ability_pool))
+
+        for item in GoofyAbility_Table.keys():
+            data = self.item_quantity_dict[item]
+            for x in range(data):
+                self.goofy_ability_pool.append(item)
+            self.item_quantity_dict.update({item: 0})
+            # 32 is the amount of goofy abilities
+        while len(self.goofy_ability_pool) < 33:
+            self.goofy_ability_pool.append(self.multiworld.per_slot_randoms[self.player].choice(self.goofy_ability_pool))
+
+    def generate_basic(self):
+        itempool: typing.List[KH2Item] = []
+
+        self.hitlist = list()
+        self.filler_items.extend(item_groups["Filler"])
+
+        if self.multiworld.FinalXemnas[self.player].value == 1:
```

Comment:
```suggestion
        if self.multiworld.FinalXemnas[self.player] == 1:
```
all the options support direct comparison, with either the option key, option name str or option int and will check that those options are even plausible

Reply:
resolved